engine/include/Engine/Intermediate/DataChunk.inl

```diff
@@ -0,0 +1,53 @@
+#include "Engine/Intermediate/DataChunk.hpp"
+
+template <typename TStoredComponent, int TSize>
+constexpr inline DataChunk<TStoredComponent, TSize>::DataChunk() noexcept
+{
+    m_components.reserve(TSize / sizeof(TStoredComponent));
+}
+
+template <typename TStoredComponent, int TSize>
+template <typename... Args>
+TStoredComponent& DataChunk<TStoredComponent, TSize>::addComponent(Args&&... args) noexcept
+{
+    std::unique_lock lock(m_mutex);
+
+    m_components.emplace_back(std::forward<Args>(args)...);
+    return m_components.back();
+}
+
+template <typename TStoredComponent, int TSize>
+void DataChunk<TStoredComponent, TSize>::destroyComponent(const TStoredComponent* componentToDestroy) noexcept
+{
+    std::unique_lock lock(m_mutex);
+
+    for (size_t i = 0; i < m_components.size(); ++i)
+    {
+        if (unlikely(&m_components[i] == componentToDestroy))
+        {
+            std::swap(m_components[i], m_components.back());
+
+            m_components[i].getGameObject().updateComponentLink<TStoredComponent>(&m_components.back(),
+                                                                                  &m_components[i]);
+            m_components.pop_back();
+        }
+    }
```

Comment:
Since a `std::vector` is contiguous in memory, we can get rid of the `for()` loop and keep a single `if`.

The content of the `if` checks that `componentToDestroy` is stored by `m_components`. It is assumed that `componentToDestroy` is not null, is a pointer to the beginning of a `TStoredComponent`, and not to the middle of one. The swap is done on `back` just because it makes the line shorter (it does not impact the associated `GameObject` negatively)

```suggestion
    // Is componentToDestroy a component in m_components?
    if (likely((componentToDestroy - m_components.data()) < (m_components.size() - 1u) * sizeof(TStoredComponent)))
    {
        const TStoredComponent& back = m_components.back();
        back->getGameObject().updateComponentLink<TStoredComponent>(&back, componentToDestroy);
        std::swap(*componentToDestroy, back);
        m_components.pop_back();
    }
```

Reply:
Done but I will only optimize this case. Each case must be checked to avoid error